VideoLibrary/videoLibrary.controller.js

```diff
@@ -0,0 +1,188 @@
+  const { _id, section, title } = req.body
+
+  if (section === "likedVideos") {
+    try {
+      videoLibrary.likedVideos.push({ _id })
+      videoLibrary = await videoLibrary.save()
+      res.status(201).json({ success: true, videoLibrary })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ success: false, message: "Couldn't be updated. Sorry!", error })
+    }
+  } else if (section === "watchLaterVideos") {
+    try {
+      videoLibrary.watchLaterVideos.push({ _id })
+      videoLibrary = await videoLibrary.save()
+      res.status(201).json({ success: true, videoLibrary })
+    } catch (error) {
+      res.status(400).json({ success: false, message: "Couldn't be updated. Sorry!", error })
+    }
+  } else if (section === "historyVideos") {
+    try {
+      videoLibrary.historyVideos.push({ _id })
+      videoLibrary = await videoLibrary.save()
+      res.status(201).json({ success: true, videoLibrary })
+    } catch (error) {
+      res.status(400).json({ success: false, message: "Couldn't be updated. Sorry!", error })
+    }
+  } else if (section === "playlistVideos") {
+    try {
+      const newPlaylist = {
+        title: title,
+        videos: [{ _id }]
+      }
+      videoLibrary.playlist.push(newPlaylist)
+      videoLibrary = await videoLibrary.save()
+      res.status(201).json({ success: true, playlist: videoLibrary.playlist })
+    } catch (error) {
+      res.status(400).json({ success: false, message: "Couldn't be updated. Sorry!", error })
+    }
+  }
+  else {
+    res.status(500).json({ success: false, message: "Invalid section provided" })
+  }
+}
+
+const deleteVideoFromVideoLibrary = async (req, res) => {
+  let { videoLibrary } = req;
+  const { _id, section } = req.body
+
+  if (section === "likedVideos") {
+    try {
+      videoLibrary.likedVideos = videoLibrary.likedVideos.filter(item => item._id != _id)
+
+      videoLibrary = await videoLibrary.save()
+      res.status(202).json({ success: true, videoLibrary })
+    } catch (error) {
+      res.status(400).json({ success: false, message: "Couldn't remove video", error })
+    }
+  } else if (section === "watchLaterVideos") {
+    try {
+      videoLibrary.watchLaterVideos = videoLibrary.watchLaterVideos.filter(item => item._id != _id)
+
+      videoLibrary = await videoLibrary.save()
+      res.status(202).json({ success: true, videoLibrary })
+    } catch (error) {
+      res.status(400).json({ success: false, message: "Couldn't remove video", error })
+    }
+  } else if (section === "historyVideos") {
+    try {
+      videoLibrary.historyVideos = videoLibrary.historyVideos.filter(item => item._id != _id)
+
+      videoLibrary = await videoLibrary.save()
+      res.status(202).json({ success: true, videoLibrary })
+    } catch (error) {
+      res.status(400).json({ success: false, message: "Couldn't remove video", error })
+    }
+  } else {
+    res.status(500).json({ success: false, message: "Invalid section provided" })
```

Comment:
Here, 400 and 500 can be interchanged. Since, message is Invalid section provided;implies its a client error, hence 400 makes more sense imo . And  for `Couldn't remove video`, 500 seems better. Let me know your thoughts